src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs

```diff
@@ -3,13 +3,206 @@
+                        yamlCodeBlockSource.Line,
+                        ex);
+                }
+            }
+        }
+
+        internal static Dictionary<string, object> ConvertContents(List<MarkdownPropertyModel> contents)
+        {
+            var contentsMetadata = new Dictionary<string, object>();
+            List<OPathSegment> OPathSegments;
+            foreach (var content in contents)
+            {
+                try
+                {
+                    OPathSegments = OverwriteUtility.ParseOPath(content.PropertyName);
+                }
+                catch (ArgumentException ex)
+                {
+                    throw new MarkdownFragmentsException(ex.Message, content.PropertyNameSource.Line, ex);
+                }
+
+                AppendNewObject(OPathSegments, content.PropertyNameSource, content.PropertyValue, contentsMetadata);
+            }
+
+            return contentsMetadata;
+        }
+
+        internal static Dictionary<string, object> MergeYamlCodeMetadataWithContentsMetadata(Dictionary<string, object> yamlCodeMetadata, Dictionary<string, object> contentsMetadata)
+        {
+            var metadata = yamlCodeMetadata.Concat(contentsMetadata).GroupBy(p => p.Key);
+            var metadatasWithSameOPath = from meta in metadata
+                where meta.Count() > 1
+                select meta;
+            foreach (var meta in metadatasWithSameOPath)
+            {
+                Logger.LogWarning(
+                    $"There are two duplicate OPaths `{meta.Key}` in yaml code block and contents block, the item in yaml code block will be overwritten by contents block item",
+                    code: WarningCodes.Overwrite.DuplicateOPaths);
+            }
+
+            return metadata.ToDictionary(g => g.Key, g => g.Last().Value);
+        }
+
+        private static void AppendNewObject(List<OPathSegment> OPathSegments, Block codeHeaderBlock, List<Block> propertyValue, Dictionary<string, object> contentsMetadata)
+        {
+            var objectValue = contentsMetadata;
+            foreach (var segment in OPathSegments)
+            {
+                if (objectValue.ContainsKey(segment.SegmentName))
+                {
+                    if (!string.IsNullOrEmpty(segment.Key))
+                    {
+                        var listObject = objectValue[segment.SegmentName] as List<Dictionary<string, object>>;
+                        if (listObject != null && listObject.Count > 0)
+                        {
+                            var goodItems = (from item in listObject
+                                where item.ContainsKey(segment.Key) &&
+                                      item[segment.Key].ToString().Equals(segment.Value.ToString())
+                                select item).ToList();
+                            if (goodItems.Count > 0)
+                            {
+                                objectValue = goodItems.First();
```

Comment:
>goodItems.First() [](start = 46, length = 17)

`goodItems[0]`?

Reply:
what is the difference?

---
In reply to: [163138314](https://github.com/dotnet/docfx/pull/2368#discussion_r163138314) [](ancestors = 163138314)